technology-brewery-commons-json/src/main/java/org/technologybrewery/commons/json/AbstractValidatedElement.java

```diff
@@ -0,0 +1,153 @@
+                                                                             TypeReference<T> typeReference) {
+
+        String fileIdentifier = jsonFile.getName();
+        try {
+            T instance = objectMapper.readValue(jsonFile, typeReference);
+            JsonNode jsonNode = objectMapper.readTree(jsonFile);
+
+            boolean valid = isValid(objectMapper, jsonNode, instance, fileIdentifier);
+            if (!valid) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug(objectMapper.writeValueAsString(instance));
+                }
+                throw new JsonException(fileIdentifier + " contained validation errors!");
+            }
+            return instance;
+
+        } catch (Exception e) {
+            throw new JsonException("Problem reading json file: " + fileIdentifier, e);
+        }
+    }
+
+    /**
+     * Useful hack to create a list of typed classes with Jackson.
+     *
+     * @param clazz the type of object in the list
+     * @return the matching {@link JavaType} for the typed list
+     */
+    static JavaType listOf(Class<?> clazz) {
+        return TypeFactory.defaultInstance().constructCollectionType(List.class, clazz);
+    }
+
+    /**
+     * Read a Json file and validate it based on the pass type.
+     *
+     * @param jsonFile     file to read
+     * @param objectMapper the Jackson {@link ObjectMapper} instance to use
+     * @param type         type to validate against
+     * @return instance of the type or a {@link Exception}
+     */
+    public static <T extends AbstractValidatedElement> List<T> readAndValidateJsonList(File jsonFile, ObjectMapper objectMapper,
+                                                                                       Class<T> type) {
+
+        String fileIdentifier = jsonFile.getName();
+        try {
+            List<T> instance = objectMapper.readValue(jsonFile, listOf(type));
+            JsonNode jsonNode = objectMapper.readTree(jsonFile);
+
+            boolean valid = isValid(objectMapper, jsonNode, instance, fileIdentifier);
+            if (!valid) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug(objectMapper.writeValueAsString(instance));
+                }
+                throw new JsonException(fileIdentifier + " contained validation errors!");
+            }
+            return instance;
+
+        } catch (Exception e) {
+            throw new JsonException("Problem reading json file: " + fileIdentifier, e);
+        }
+    }
+
+    private static <T extends AbstractValidatedElement> boolean isValid(ObjectMapper objectMapper, JsonNode jsonInstance,
+                                                                        List<T> instance, String fileIdentifier) throws JsonException {
+        boolean isValid = false;
+        T fistInstance;
+        if (instance != null && instance.isEmpty()) {
```

Comment:
Q/S: Is think this supposed to be `!instance.isEmpty()`?

Reply:
✅ Yep - it is.  I had fixed that locally, but forgot to push it up.  Good catch and fixed!